tq/basic_upload.go

```diff
@@ -86,7 +86,7 @@ func (a *basicUploadAdapter) DoTransfer(ctx interface{}, t *Transfer, cb Progres
 		return nil
 	}
 
-	cbr := tools.NewBodyWithCallback(f, t.Size, ccb)
+	cbr := tools.NewFileBodyWithCallback(f, t.Size, ccb)
```

Comment:
This seems sane.  We defer `f.Close()` up above, so we won't leak the file descriptor.

Reply:
Test added.